src/spaced/spaced/game/damage.hpp

```diff
@@ -1,17 +1,27 @@
 #pragma once
-#include <bave/core/time.hpp>
-#include <glm/vec2.hpp>
+#include <bave/imgui/im_text.hpp>
+#include <bave/app.hpp>
 
 namespace spaced {
 class Damage {
   public:
-	void inflict(float amount);
-	bool is_dead() const;
+	constexpr void inflict(int amount) { m_health -= glm::clamp(amount, amount, m_health); }
+	constexpr bool is_dead() const { return m_health <= 0; }
+	constexpr int get_health() const { return m_health; }
 
-  private:
-	void update();
+	void debug_stuff() {
+		if (ImGui::TreeNode("damage")) {
+			bave::im_text("health: {}", m_health);
+			bave::im_text("dead: {}", is_dead() ? "yes" : "no");
+			bool inflict_ten{};
+			ImGui::Checkbox("inflict 10 damage", &inflict_ten);
+			if (inflict_ten) { inflict(10); }
+			ImGui::SliderInt("health", &m_health, 0, 100);
+			ImGui::TreePop();
+		}
+	}
```

Comment:
A couple of points:
1. All ImGui code needs to be within `if constexpr (bave::imgui_v)` blocks, otherwise the build will fail on Android.
2. This _does_ belong in a cpp file since it needs huge dependencies that aren't required in the header (ImGui related).